src/lib.rs

```diff
@@ -0,0 +1,1210 @@
+    type Item = bool;
+    type IntoIter = Iter<'a, T>;
+
+    fn into_iter(self) -> Iter<'a, T> {
+        self.iter()
+    }
+}
+
+#[derive(Clone)]
+pub struct IterSetBits<'a, T: 'a> {
+    vob: &'a Vob<T>,
+    range: Range<usize>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterSetBits<'a, T> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<usize> {
+        if let Some(mut i) = self.range.next() {
+            for b in block_offset::<T>(i) .. blocks_required::<T>(self.vob.len) {
+                let v = self.vob.vec[b];
+                if v != T::zero() {
+                    let i_off = i % bits_per_block::<T>();
+                    let tz = (v >> i_off).trailing_zeros() as usize;
+                    if tz < bits_per_block::<T>() {
+                        let bs = b * bits_per_block::<T>() + i_off + tz;
+                        self.range.start = bs + 1;
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+#[derive(Clone)]
+pub struct IterUnsetBits<'a, T: 'a> {
+    vob: &'a Vob<T>,
+    range: Range<usize>,
+}
+
+impl<'a, T: Debug + One + PrimInt + Zero> Iterator for IterUnsetBits<'a, T> {
+    type Item = usize;
+
+    fn next(&mut self) -> Option<usize> {
+        if let Some(mut i) = self.range.next() {
+            for b in block_offset::<T>(i) .. blocks_required::<T>(self.vob.len) {
+                let v = self.vob.vec[b];
+                if v != T::max_value() {
+                    let i_off = i % bits_per_block::<T>();
+                    let tz = (!v >> i_off).trailing_zeros() as usize;
+                    if tz < bits_per_block::<T>() {
+                        let bs = b * bits_per_block::<T>() + i_off + tz;
+                        self.range.start = bs + 1;
+                        if bs >= self.vob.len {
+                            break;
+                        }
+                        return Some(bs);
+                    }
+                }
+                i = b * bits_per_block::<T>();
+            }
+        }
+        self.range.start = self.range.end;
+        None
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.range.size_hint()
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> PartialEq for Vob<T> {
+    fn eq(&self, other: &Self) -> bool {
+        if self.len != other.len {
+            return false;
+        }
+        self.iter_storage()
+            .zip(other.iter_storage())
+            .all(|(w1, w2)| w1 == w2)
+    }
+}
+
+impl<T: Debug + One + PrimInt + Zero> Eq for Vob<T> {}
+
+impl<T :Debug + Hash + One + PrimInt + Zero> Hash for Vob<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.len.hash(state);
```

Comment:
Was there an informed decision behind including the length in the hash?

I can't decide if including it would lead to fewer collisions...

Reply:
I think I was influenced by `Vec`. But I agree: it does seem odd. Let's kill it, and if we've done something daft, someone will tell us...